pkg/bpf/map_linux.go

```diff
@@ -452,6 +452,16 @@ func (m *Map) OpenOrCreateUnpinned() (bool, error) {
 	return m.openOrCreate(false)
 }
 
+// Create is similar to OpenOrCreate, but closes the map after creating or
+// opening it.
+func (m *Map) Create() (bool, error) {
+	isNew, err := m.OpenOrCreate()
+	if err != nil {
+		return isNew, err
+	}
+	return isNew, m.Close()
```

Comment:
For reviewers: this can be rewritten to avoid a open(2) and close(2) when the map already exists. Is that worth the additional code?

Reply:
but won't the maps be open again in the future by the daemon?

Reply:
Not all of them. Some of not used by the cilium-agent (e.g., neighbor maps for BPF-based NodePort). The goal here is to make sure they're all created before any BPF program is loaded. I think the longer-term goal is to perform all map creations + program loading on the Go side (i.e., remove the tc dependency).

Reply:
I doubt that we'll see much impact from a couple of dozen syscalls occuring on startup, so I'm fine with deferring this for now.